test_runner/regress/test_pageserver_generations.py

```diff
@@ -205,6 +205,9 @@ def test_generations_upgrade(neon_env_builder: NeonEnvBuilder):
         sk.start()
     env.attachment_service.start()
 
+    # We will start a pageserver with no control_plane_api set, so it won't be able to self-register
+    env.attachment_service.node_register(env.pageserver)
```

Comment:
Can we move this to the place below where we enable control_plane_api in the PS config? Would be better story-telling because this test demonstrates the upgrade path.

Reply:
This needs to be at the top: otherwise things like create_tenant break (they go via the storage controller).